### PR TITLE
build/spec/systemd fixups

### PR DESCRIPTION
**Pull request by alonbl**
Hi,
These are fixes of the build/spec/systemd.
Superset of #99.

@sketch242: I have no fedora/rhel environment, please check if working as expected.

Thanks!

### CODE REVIEW

**Comment by alonbl** [comment]
@philwhineray: please review.

File: system/Makefile.am [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+MAINTAINERCLEANFILES= $(srcdir)/Makefile.in
+CLEANFILES = \
+	netdata.service \
+	$(NULL)
+
+include $(top_srcdir)/build/subst.inc
+
+SUFFIXES = .in
+
+dist_noinst_DATA = \
```
Comment (philwhineray) [inline system/Makefile.am]: As written the user will get a `netdata.service` for the configuration we package, not one appropriate to their configuration.

I guess this should be made noinst_DATA and then add EXTRA_DIST= netdata-openrc netdata.service.in, or something equivalent.

**Comment by philwhineray** [comment]
Just the one comment regarding `system/Makefile.am`. Other than that it looks OK to me.

File: system/Makefile.am (hunk 1 above)
Comment (alonbl) [inline system/Makefile.am]: there is no major difference between EXTRA_DIST and dist_noinst_DATA, I prefer to use the same method for both, please notice that both are not installed.

Comment (philwhineray) [inline system/Makefile.am]: The difference here is we want to package the `netdata.service.in`, not the `netdata.service` I think.

Comment (alonbl) [inline system/Makefile.am]: correct! sorry for that. will push a fix soon.

**Comment by philwhineray** [comment]
Looks OK to me now. I say merge it.

**Comment by alonbl** [comment]
> Looks OK to me now. I say merge it.

I still need confirmation from @sketch242 rhel/fedora/systemd spec file is ok and working.

**Comment by sketch242** [comment]
@alonbl Not quite.  I just submitted a pull request against your repo...

**Comment by alonbl** [comment]
@sketch242: thanks, I've added it with minor modification, can you please recheck, we should be good now.

**Comment by alonbl** [comment]
@sketch242: now? :)

**Comment by sketch242** [comment]
@alonbl One more very minor change to the build, which I see you already got.

Just to clarify, there still seem to be two issues with the service actually working:

* The webserver checks permissions of /usr/share/netdata unnecessarily and doesn't work with the directory being owned by root.  I'm not sure if @ktsaou is fixing that or not.
* Even if I fix the above manually, the service still doesn't start successfully...but running netdata manually works fine.  Not sure why, this didn't happen with the 1.0.0 release, and the only change to the systemd unit file is the PID file location (which seems to be ignored anyway?), and changing that does not alter the behavior.

**Comment by alonbl** [comment]
Hi,
1. These two issues are unrelated to this change. the service must run when /usr/share/netdata is owned by root, if this is not the case it is a bug.
2. the entire service implementation should be fixed so it won't be forking daemon, so it will be well behaved to both openrc and systemd. If it won't fork, then pid should not be used at all at service level (openrc start-stop-deamon will create its own).
Thanks!

**Comment by alonbl** [comment]
merged these, please keep working on these two issues.

**Comment by ktsaou** [comment]
@alonbl I think both the issues you are mentioning are already solved.

1. You can start netdata as any user and it won't attempt to switch users.
2. You can pass `-nd` and the daemon will not fork.

What there is to work on?

**Comment by alonbl** [comment]
@ktsaou:
1. I remember that there was an issue with running as non root, you wrote once that you need somekind of capability. Most of these can be provided by system service keywords, so if that true then you drop the root usage.
2. I see that pid file is created unconditionally, you should add a parameter for a service --pidfile or similar with a location of pidfile to create, if not given, do not create one. Then modify the systemd/openrc to run the service with no forking and no pidfile.

**Comment by ktsaou** [comment]
ok, I'll open issues for both of them to track them down.

Regarding the capabilities, do you know if the issue at https://github.com/firehol/netdata/issues/149#issuecomment-204838428 can be solved with capabilities?

**Comment by alonbl** [comment]
@ktsaou: I think we discussed that in the past. you should split this into own service and communicate using usock or similar mechanism. so you run your netdata as netdata user and you run the proc monitor as root and communicate using usock that is rw by netdata.

**Comment by ktsaou** [comment]
> I think we discussed that in the past. you should split this into own service and communicate using usock or similar mechanism. so you run your netdata as netdata user and you run the proc monitor as root and communicate using usock that is rw by netdata.

This is what netdata already does. Only `apps.plugin` runs as root.

**Pull request merged**